oxide-auth-db/src/db_service/redis.rs

```diff
@@ -0,0 +1,152 @@
+use r2d2_redis::RedisConnectionManager;
+use r2d2::Pool;
+use std::env;
+use r2d2_redis::redis::Commands;
+use oxide_auth::primitives::registrar::{EncodedClient, ClientType};
+use crate::primitives::db_registrar::OauthClientDBRepository;
+use dotenv::dotenv;
+use url::Url;
+use oxide_auth::primitives::prelude::Scope;
+use std::str::FromStr;
+
+pub const REDIS_POOL_SIZE: u32 = 32;
+
+/// redis datasource to Client entries.
+
+#[derive(Debug, Clone)]
+pub struct RedisDataSource {
+    pub url: String,
+    pub pool: Pool<RedisConnectionManager>,
+}
+
+impl RedisDataSource {
+
+    pub fn new() -> Self {
+        dotenv().ok();
+        let url = env::var("REDIS_URL").expect("REDIS_URL must be set");
+        let pool = REDIS_POOL.clone();
+        RedisDataSource { url, pool }
+    }
```

Comment:
I don't feel super comfortable defining and requiring a single static to control a library's behaviour, especially if it modifies the global environment in the constructor. I'd rather take a reference to the pool as a parameter here and leave it to the user.

Reply:
Thanks for the advise. I have refatored these.